src/paperoni/refinement/html/html.py

```diff
@@ -0,0 +1,99 @@
+import hashlib
+from typing import Literal
+
+from outsight import send
+from paperazzi.platforms.utils import Message
+from requests import HTTPError
+
+from ...config import config
+from ...model.classes import Author, Institution, Link, Paper, PaperAuthor
+from ...prompt import ParsedResponseSerializer
+from ..fetch import register_fetch
+from .model import FIRST_MESSAGE, SYSTEM_MESSAGE, Analysis
+
+
+def _make_key(_, kwargs: dict) -> str:
+    kwargs = kwargs.copy()
+    kwargs["messages"] = kwargs["messages"][:]
+    for i, message in enumerate(kwargs["messages"]):
+        message: Message
+        if message.prompt == FIRST_MESSAGE:
+            # Use only the link hash to compute the key. We do not expect the
+            # html_content to change much so most of the time we will not need
+            # to re-run the prompt.
+            kwargs["messages"][i] = Message(
+                type=message.type,
+                prompt=message.prompt,
+            )
+
+    return config.refine.prompt._make_key(None, kwargs)
+
+
+def prompt(link: str, force: bool = False) -> Analysis:
+    try:
+        html_content = config.fetch.read(f"https://doi.org/{link}", format="txt")
+    except HTTPError as exc:  # pragma: no cover
+        if exc.response.status_code == 404:
+            return None
+        else:
+            raise
```

Comment:
Is this the only doi url we should check? I wasn't sure where to look for a list of possible doi urls that are supported

Reply:
ah yes sure that makes a lot more sense